src/renate/benchmark/datasets/nlp_datasets.py

```diff
@@ -1,78 +1,120 @@
+    """Data module wrapping Huggingface text datasets.
+
+    This is convenience wrapper to expose a hugginface dataset as a `RenateDataModule`. Datasets
+    will be pre-tokenized and will return `input, target = dataset[i]`, where `input` is a
+    dictionary with fields `["input_ids", "attention_mask"]`, and `target` is a tensor.
+
+    We expect the dataset to have a "train" and a "test" split. An additional "validation" split
+    will be used if present. Otherwise, a validation set may be split off of the training data
+    using the `val_size` argument.
 
     Args:
         data_path: the path to the folder containing the dataset files.
-        src_bucket: the name of the s3 bucket. If not provided, downloads the data from original
-            source.
-        src_object_name: the folder path in the s3 bucket.
-        dataset_name: Name of the torchvision dataset.
+        dataset_name: Name of the dataset, see https://huggingface.co/datasets.
+        input_column: Name of the column containing the input text.
+        target_column: Name of the column containing the target (e.g., class label).
+        tokenizer: Tokenizer to apply to the dataset.
+        tokenize_kwargs: Keyword arguments to be passed to the tokenizer.
         val_size: Fraction of the training data to be used for validation.
         seed: Seed used to fix random number generation.
     """
 
-    dataset_dict = {
-        "AG_NEWS": (torchtext.datasets.AG_NEWS, "ag_news_csv"),
-        "AmazonReviewFull": (torchtext.datasets.AmazonReviewFull, "amazon_review_full_csv"),
-        "DBpedia": (torchtext.datasets.DBpedia, "dbpedia_csv"),
-    }
-
     def __init__(
         self,
-        data_path: Union[Path, str],
-        src_bucket: Optional[str] = None,
-        src_object_name: Optional[str] = None,
-        dataset_name: str = "AG_NEWS",
+        data_path: str,
+        dataset_name: str = "ag_news",
+        input_column: str = "text",
+        target_column: str = "label",
+        tokenizer: Optional[transformers.PreTrainedTokenizer] = None,
+        tokenizer_kwargs: Optional[dict] = None,
         val_size: float = defaults.VALIDATION_SIZE,
         seed: int = defaults.SEED,
     ):
-        super(TorchTextDataModule, self).__init__(
-            data_path,
-            src_bucket=src_bucket,
-            src_object_name=src_object_name,
+        super(HuggingfaceTextDataModule, self).__init__(
+            data_path=data_path,
             val_size=val_size,
             seed=seed,
         )
-        if dataset_name not in TorchTextDataModule.dataset_dict:
-            raise ValueError(f"Dataset {self._dataset_name} currently not supported.")
         self._dataset_name = dataset_name
+        self._input_column = input_column
+        self._target_column = target_column
+        self._tokenizer = tokenizer
+        self._tokenizer_kwargs = tokenizer_kwargs or dict(
+            padding="max_length", max_length=128, truncation=True
```

Comment:
it would be good to move the default configuration of the tokenizer in the Renate defaults.

Reply:
I don't think that's a good idea, since this is benchmarking code that has nothing to do with renate.

Reply:
I don't see that as a problem, but we can create a set of default for the benchmarks if you prefer

Reply:
These defaults are specific to huggingface tokenizers and will only be used here. Is there really a good reason to outsource this to a separate file? If I'm a user of this class, I want to _see_ the defaults so that I can decide whether I need to change them.

Reply:
Moved it to `defaults.py` after offline discussion